tofu/modules/il_gcc_application/main.tf

```
@@ -0,0 +1,218 @@
+        app_key                    = ""
+        session_replay_sample_rate = ""
+        rum_app_id                 = ""
+        rum_client_token           = ""
+        environment                = ""
+      })
+    },
+    "jobrunr" = {
+      description     = "JobRunr configuration for the IL-GCC application."
+      recovery_window = var.secret_recovery_period
+      start_value = jsonencode({
+        dashboard_enabled      = ""
+        enable_background_jobs = ""
+      })
+    },
+    "ccms" = {
+      description     = "CCMS credentials for the IL-GCC application."
+      recovery_window = var.secret_recovery_period
+      start_value = jsonencode({
+        ocp_apim_key       = ""
+        api_base_url       = ""
+        api_username       = ""
+        api_password       = ""
+        transaction_delay  = ""
+        enable_integration = ""
+      })
+    },
+    "il-gcc" = {
+      description     = "IL-GCC application configuration."
+      recovery_window = var.secret_recovery_period
+      start_value = jsonencode({
+        wait_for_provider_response  = ""
+        allow_provider_registration = ""
+        convert_uploads_to_pdf      = ""
+        enable_dts_integration      = ""
+        spring_profiles_active      = ""
+      })
+    }
+  }
+
+  tags = { service = "il-gcc-application" }
+}
+
+module "database" {
+  source = "github.com/codeforamerica/tofu-modules-aws-serverless-database?ref=1.0.0"
+
+  logging_key_arn = var.logging_key
+  secrets_key_arn = module.secrets.kms_key_arn
+  vpc_id          = var.vpc_id
+  subnets         = var.private_subnets
+  ingress_cidrs   = ["10.0.28.0/22"]
+  force_delete    = var.force_delete
+
+  min_capacity        = var.database_capacity_min
+  max_capacity        = var.database_capacity_max
+  skip_final_snapshot = var.database_skip_final_snapshot
+  apply_immediately   = var.database_apply_immediately
+  key_recovery_period = var.key_recovery_period
+  snapshot_identifier = var.database_snapshot
+
+  project     = "illinois-getchildcare"
+  environment = var.environment
+  service     = "app"
+
+  tags = { service = "app" }
+}
+
+# Deploy the IL-GCC Application to a Fargate cluster.
+module "service" {
+  source = "github.com/codeforamerica/tofu-modules-aws-fargate-service?ref=1.1.0"
+  project                = "illinois-getchildcare"
+  project_short          = "il-gcc"
+  environment            = var.environment
+  service                = "app"
+  service_short          = "app"
+  domain                 = var.domain
+  subdomain              = var.subdomain
+  vpc_id                 = var.vpc_id
+  private_subnets        = var.private_subnets
+  public_subnets         = var.public_subnets
+  logging_key_id         = var.logging_key
+  container_port         = 8080
+  force_delete           = var.force_delete
+  create_endpoint        = true
+  image_tags_mutable     = true
+  image_tag              = data.aws_ssm_parameter.version.insecure_value
```

When updating to `1.2.0` you can pass in the name of the SSM parameter instead of having to look it up yourself. You can replace `image_tag` with `aws_ssm_parameter` and remove the `data "aws_ssm_parameter" "version"` block on line 12. 

```suggestion
  aws_ssm_parameter      = aws_ssm_parameter.version.name
```

Optionally, you can replace the SSM parameter you've created entirely and use `create_version_parameter` to have this module create and manage the parameter. This would require updating your deployment pipelines to point at the newly created parameter.

If you choose to go this route, you can set `image_tag` to the current version of your SSM parameter to pre-populate the new parameter (`image_tag` will be ignored after that), otherwise it will be set to `latest` initially. 

```suggestion
  image_tag                            = "abc123"
  create_version_parameter = true
```

You could then remove the `resource "aws_ssm_parameter" "version"` block on line 1, in addition to the data block.

---

I'm not quite sure what the latter part of this is saying to do, but it sounds like it would be optimal? I made the initial change from `image_tag` to `aws_ssm_parameter` but maybe we could pair on the latter change if you have some free time in the coming week?

---

Absolutely! The latter part isn't a requirement so it shouldn't block you, but happy to pair on it.